website/src/content/docs/docs/extending-typespec/emitter-framework.md

```diff
@@ -3,369 +3,118 @@ id: emitter-framework
-
-> declaration of Pet with a property named position and a type of an object literal
-
-#### References between emitted types
-
-A common scenario when emitting to most targets is handling how to make references between types. This can get pretty complex, especially when the declarations are emitted into different scopes. The emitter framework does a lot of heavy lifting for you by calculating the scopes between your current context and the declaration you're trying to reference.
-
-How declarations arrive in different scopes is covered in the Context section later in this document.
-
-Let's look at the `reference` signature on the TypeEmitter:
-
-```typescript
-  reference(
-    targetDeclaration: Declaration<string>,
-    pathUp: Scope<string>[],
-    pathDown: Scope<string>[],
-    commonScope: Scope<string> | null
-  ): string | EmitEntity<string> {}
-```
-
-The `reference` function is called with:
-
-- `targetDeclaration`: The declaration we're making a reference to.
-- `pathUp`: The scopes between our current scope and the common scope.
-- `pathDown`: The scopes between the common scope and the declaration we're referencing.
-- `commonScope`: The nearest scope shared between our current scope and the target declaration.
-
-So let's imagine we have declarations under the following scopes:
-
-```
-source file
-  namespace A
-    namespace B
-      model M1
-  namespace C
-    model M2
-```
-
-If M1 references M2, `reference` will be called with the following arguments:
-
-- `targetDeclaration`: M2
-- `pathUp`: [namespace B, namespace A]
-- `pathDown`: [namespace C]
-- `commonScope`: source file
-
-For languages which walk up a scope chain in order to find a reference (e.g. TypeScript, C#, Java, etc.), you generally won't need `pathUp`, you can just join the scopes in the `pathDown` resulting in a reference like `C.M2`. Other times you may need to construct a more path-like reference, in which case you can emit for example a `../` for every item in `pathUp`, resulting in a reference like `../../C/M2`.
-
-When the declarations don't share any scope, `commonScope` will be null. This happens when the types are contained in different source files. In such cases, your emitter will likely need to import the target declaration's source file in addition to constructing a reference. The source file has an `imports` property that can hold a list of such imports.
-
-We can update our example emitter to generate references by adding an appropriate `references` method:
-
-```typescript
-class MyCodeEmitter extends CodeTypeEmitter {
-  // snip out the methods we implemented previously
-
-  // If the model is Person, put it into a special namespace.
-  // We will return to this in detail in the next section.
-  modelDeclarationContext(model: Model, name: string): Context {
-    if (model.name === "Person") {
-      const parentScope = this.emitter.getContext().scope;
-      const scope = this.emitter.createScope({}, "Namespace", parentScope);
+::: caution
```

Comment:
```suggestion
:::caution
```

i think, it doesn't render well now 
<img width="784" alt="image" src="https://github.com/user-attachments/assets/c28baf59-3628-4f78-8dfb-3e2fba42f358" />